pyensign/utils/rlid.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any
+
+
+RLID_SIZE = 10
+ENCODED_SIZE = 16
+ENCODING = "0123456789abcdefghjkmnpqrstvwxyz"
+
+
+class RLID:
+    """
+    An RLID is a 10 byte totally ordered unique lexicographically sortable identifier.
+    In string format, they are 16 characters long. RLIDs are usually generated by the
+    Ensign server and not by the pyensign SDK.
+    """
+
+    def __init__(self, bytes):
+        if len(bytes) != RLID_SIZE:
+            raise ValueError("Unexpected RLID size: {}".format(len(bytes)))
+        self.bytes = bytes
+
+    def __repr__(self):
+        return "RLID({})".format(self.bytes)
+
+    def __str__(self):
+        """
+        Port of the RLID encoding method from the Go SDK.
+        """
+        dst = [None] * ENCODED_SIZE
+        dst[0] = ENCODING[(self.bytes[0] & 248) >> 3]
+        dst[1] = ENCODING[((self.bytes[0] & 7) << 2) | ((self.bytes[1] & 192) >> 6)]
+        dst[2] = ENCODING[(self.bytes[1] & 62) >> 1]
+        dst[3] = ENCODING[((self.bytes[1] & 1) << 4) | ((self.bytes[2] & 240) >> 4)]
+        dst[4] = ENCODING[((self.bytes[2] & 15) << 1) | ((self.bytes[3] & 128) >> 7)]
+        dst[5] = ENCODING[(self.bytes[3] & 124) >> 2]
+        dst[6] = ENCODING[((self.bytes[3] & 3) << 3) | ((self.bytes[4] & 224) >> 5)]
+        dst[7] = ENCODING[self.bytes[4] & 31]
+        dst[8] = ENCODING[(self.bytes[5] & 248) >> 3]
+        dst[9] = ENCODING[((self.bytes[5] & 7) << 2) | ((self.bytes[6] & 192) >> 6)]
+        dst[10] = ENCODING[(self.bytes[6] & 62) >> 1]
+        dst[11] = ENCODING[((self.bytes[6] & 1) << 4) | ((self.bytes[7] & 240) >> 4)]
+        dst[12] = ENCODING[((self.bytes[7] & 15) << 1) | ((self.bytes[8] & 128) >> 7)]
+        dst[13] = ENCODING[(self.bytes[8] & 124) >> 2]
+        dst[14] = ENCODING[((self.bytes[8] & 3) << 3) | ((self.bytes[9] & 224) >> 5)]
+        dst[15] = ENCODING[self.bytes[9] & 31]
+        return "".join(dst)
```

Comment:
This is ported from the Go SDK, we don't have any need to be able to decode or create RLIDs from pyensign yet (expect in maybe tests) so this is purely for displaying them nicely at the moment.

Reply:
Ok, in the future if we do need that ability in Python we can move this to `encode`/`decode` methods.